src/main/java/com/gtnewhorizons/angelica/api/QuadProvider.java

```diff
@@ -0,0 +1,15 @@
+package com.gtnewhorizons.angelica.api;
+
+import com.gtnewhorizons.angelica.compat.mojang.BlockPos;
+import com.gtnewhorizons.angelica.compat.nd.Quad;
+import com.gtnewhorizons.angelica.utils.ObjectPooler;
+import java.util.List;
+import java.util.Random;
+import net.minecraft.block.Block;
+import net.minecraft.world.IBlockAccess;
+import net.minecraftforge.common.util.ForgeDirection;
+
+public interface QuadProvider {
+
+    List<Quad> getQuads(IBlockAccess world, BlockPos pos, Block block, int meta, ForgeDirection dir, Random random, ObjectPooler<Quad> quadPool);
```

Comment:
Do these pooled quads ever get released?  If not the pool will just grow unbounded.

Reply:
The pool only stores quads waiting to be used - constantly getting quads is the same as just allocating. The case where you get a growing pool is if you constantly release quads into it. I was going to handle that in a later PR, because if I do that I have to make CapturingTessellator use the right pool too, or it'll continuously dump quads into Blockrenderer's pool and make an actual memory leak.

Reply:
Sure.  But constantly allocating would get released when it goes out of scope.  Here it never will without a release

Reply:
It goes out of scope in renderQuad, no? ObjectPooler only keeps references to free objects in its pool. If you never `releaseInstance()` it the GC picks it up eventually.

Reply:
Added a release in BlockRenderer